javascript/ql/src/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -220,11 +220,14 @@ abstract class FlowLabel extends string {
   bindingset[this] FlowLabel() { any() }
 }
 
-module FlowLabel {
-  private class StandardFlowLabel extends FlowLabel {
-    StandardFlowLabel() { this = "data" or this = "taint" }
-  }
+/**
+ * A standard flow label, that is, either `FlowLabel::data()` or `FlowLabel::taint()`.
+ */
+private class StandardFlowLabel extends FlowLabel {
+  StandardFlowLabel() { this = "data" or this = "taint" }
```

Comment:
Minor design issue on internal class: shouldn't this be using `FlowLabel::data()`  and `FlowLabel::taint()` to hide the string implementation detail?

Reply:
That would lead to an empty recursion due to type-joining on the result type of `FlowLabel::data()` and `FlowLabel::taint()`.